kafka-ui-api/src/main/java/com/provectus/kafka/ui/serdes/builtin/ProtobufFileSerde.java

```diff
@@ -127,6 +159,17 @@ void configure(
     this.keyMessageDescriptorMap = keyMessageDescriptorMap;
   }
 
+  @Nullable
+  private static Location toLocation(Path path) {
+    if (Files.isDirectory(path)) {
+      return Location.get(path.toString());
+    }
+    if (Files.isRegularFile(path)) {
+      return Location.get(path.getParent().toString(), path.getFileName().toString());
+    }
+    return null;
+  }
```

Comment:
just a question: so now we can set directory with `.proto` files? If so, will it be scanned recursively?

Reply:
@iliax Yes, that's correct. Directories will be scanned recursively.

The Square Wire library even supports reading Protobuf schemas from ZIP (and JAR) archives.

I have updated the documentation and added explicit support for reading Protobuf files from archives, too.

➡️ https://github.com/provectus/kafka-ui/pull/3039/commits/9b226e3fe9aebe701307bf341db1f07440ef9a9e